From b5413512dd22282f174aadb4c0b78f5117d68682 Mon Sep 17 00:00:00 2001
Message-Id: <b5413512dd22282f174aadb4c0b78f5117d68682.1373885145.git.jdenemar@redhat.com>
From: Martin Kletzander <mkletzan@redhat.com>
Date: Mon, 8 Jul 2013 09:26:44 +0200
Subject: [PATCH] qemu: Don't miss errors when changing graphics passwords

https://bugzilla.redhat.com/show_bug.cgi?id=975751

Commit 23e8b5d8e7a92bac85b7fd2aca8992501bf680ee forgot to check the
return value for all calls to qemuDomainChangeGraphicsPasswords().
(cherry picked from commit 556808ec9deae8623aea2a3a8c1c80a62a335c4a)
---
 src/qemu/qemu_process.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/qemu/qemu_process.c b/src/qemu/qemu_process.c
index aa561d1..bb16112 100644
--- a/src/qemu/qemu_process.c
+++ b/src/qemu/qemu_process.c
@@ -2172,10 +2172,10 @@ qemuProcessInitPasswords(virConnectPtr conn,
                                                     &graphics->data.spice.auth,
                                                     driver->spicePassword);
         }
-    }
 
-    if (ret < 0)
-        goto cleanup;
+        if (ret < 0)
+            goto cleanup;
+    }
 
     if (qemuCapsGet(priv->caps, QEMU_CAPS_DEVICE)) {
         int i;
-- 
1.8.3.2

